x-pack/plugin/sql/sql-action/src/test/java/org/elasticsearch/xpack/sql/action/SqlRequestParsersTests.java

```diff
@@ -119,12 +127,16 @@ public void testQueryRequestParser() throws IOException {
         
         SqlQueryRequest request = generateRequest("{\"cursor\" : \"whatever\", \"mode\" : \""
                 + randomMode.toString() + "\", \"client_id\" : \"bla\","
+                + clientVersion
                 + "\"query\":\"select\","
                 + "\"params\":[" + params + "],"
                 + " \"time_zone\":\"UTC\","
                 + "\"request_timeout\":\"5s\",\"page_timeout\":\"10s\"}", SqlQueryRequest::fromXContent);
         assertNull(request.clientId());
         assertEquals(randomMode, request.mode());
+        if (clientVersion.isEmpty() == false) {
```

Comment:
Why not using the same condition as above? `Mode.isDedicatedClient(randomMode)`

Reply:
good point.